### ISSUE TITLE
Key Not Found Exeption: The given key was not present in the dictionary

### ISSUE THREAD

**Issue by narqs** [omansak/libvideo#27]
Hey im having problem with some urls
Here is one of them: https://www.youtube.com/watch?v=B3eAMGXFw1o
Keep getting a key not found exception, it points to ServiceBase.cs  
Line 21: GetVideoAsync(videoUri).GetAwaiter().GetResult();
is anyone getting this too?

**Comment by rubensollie** [comment]
Could it be, because of age blocking?

On youtube.com it will ask you for an age, before it loads the video. libvideo probably can't find the url's from the page, because they are not available yet, because of the age popup.

**Comment by narqs** [comment]
Oh did not notice that...

**Comment by rubensollie** [comment]
So, this seems to be the reason why: 

"Unfortunately, this video is not available in your country because it could contain music, for which we could not agree on conditions of use with GEMA. 

Sorry about that."

EDIT: Nevermind, YouTube thought I was in Germany, because of VPN :)
It is definitely the cause of the age popup. Going to think of a way to get past it :P

**Comment by jamesqo** [comment]
Hey guys, sorry for not responding, been a little busy lately.

@rubensollie is most likely correct, the issue is probably due to the age restrictions on the video. I'll take a look into it when I have time and get back to you guys on this. It would be great if someone could submit a PR though :wink:

**Comment by barterajay** [comment]
https://www.youtube.com/watch?v=GdBaljNAJbE for this i am not getting url in live but getting url in debugging mode... in live exception is same as yours.. any idea??

**Comment by ATypescriptEnjoyer** [comment]
+1 with this. I was trying every Catfish and the bottlemen song :( some arent age rating'd

**Comment by soapergem** [comment]
@jamesqo I submitted a PR yesterday which should fix this issue.

**Comment by gogognome** [comment]
For the URL https://www.youtube.com/watch?v=kp8u_Yrw76Q I also get the KeyNotFoundException. Turns out that adaptive_fmts returned no queries, which resulted again in Unscramble being called with an empty string as parameter. I will submit a Pull Request to fix this.

**Comment by soapergem** [comment]
@gogognome I already submitted one...

**Comment by gogognome** [comment]
@soapergem I first only saw your comment about adding an if and I thought that wouldn't solve the problem I faced. I just tested URL https://www.youtube.com/watch?v=kp8u_Yrw76Q on your pull request and I see that Unscramble gets called with queryStrings containing only the empty string. The empty string does not contain the URL, so the if statement prevents the KeyNotFoundException. My solution prevents Unscramble from ever getting called with an empty string.

Both pull requests fix the KeyNotFoundException for the aforementioned URL. Now jamesqo can choose from two different solutions. :-)
